packages/docusaurus-plugin-content-blog/src/authorsMap.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import _ from 'lodash';
+import {getDataFileData, normalizeUrl} from '@docusaurus/utils';
+import {Joi, URISchema} from '@docusaurus/utils-validation';
+import type {BlogContentPaths} from './types';
+import type {
+  Author,
+  AuthorAttributes,
+  AuthorPage,
+} from '@docusaurus/plugin-content-blog';
+
+export type AuthorsMap = {[authorKey: string]: Author};
+
+type AuthorInput = AuthorAttributes & {
+  page: boolean | AuthorPage;
+};
+
+type AuthorsMapInput = {[authorKey: string]: AuthorInput};
+
+const AuthorPageSchema = Joi.object<AuthorPage>({
+  permalink: Joi.string().required(),
+});
+
+const AuthorsMapInputSchema = Joi.object<AuthorsMapInput>()
+  .pattern(
+    Joi.string(),
+    Joi.object({
+      name: Joi.string(),
+      url: URISchema,
+      imageURL: URISchema,
+      title: Joi.string(),
+      email: Joi.string(),
+      page: Joi.alternatives(Joi.bool(), AuthorPageSchema),
+      permalink: Joi.string(),
```

Comment:
We allow bool or AuthorPageSchema, however the type definition of page is null or AuthorPageSchema, tests throws page should be boolean or AuthorPage, else TS yells page to be null

Reply:
I'm not sure to understand what you mean here 😅 

I think I forgot the `.try()` in `.alternatives().try()`, maybe that's why it doesn't work well?